onmt/Trainer.py

```diff
@@ -0,0 +1,166 @@
+import torch
+import onmt
+import onmt.modules
+
+
+class Trainer(object):
+    def __init__(self, model, train_data, valid_data, fields, optim,
+                 batch_size, gpuid, copy_attn, copy_attn_force,
+                 truncated_decoder, max_generator_batches):
+        # Basic attributes.
+        self.model = model
+        self.train_data = train_data
+        self.valid_data = valid_data
+        self.fields = fields
+        self.optim = optim
+        self.batch_size = batch_size
+        self.gpuid = gpuid
+        self.copy_attn = copy_attn
+        self.truncated_decoder = truncated_decoder
```

Comment:
Oh sorry, this is a bad name. It is an integer, so name should indicate it is a length.